recipes/grpc/all/conanfile.py

```diff
@@ -0,0 +1,161 @@
+                raise ConanInvalidConfiguration("gRPC can only be built with Visual Studio 2015 or higher.")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+
+        # This doesn't work yet as one would expect, because the install target builds everything
+        # and we need the install target because of the generated CMake files
+        #
+        #   enable_mobile=False # Enables iOS and Android support
+        #
+        # cmake.definitions["CONAN_ENABLE_MOBILE"] = "ON" if self.options.build_csharp_ext else "OFF"
+
+
+        cmake.definitions["gRPC_BUILD_CODEGEN"] = "ON" if self.options.build_codegen else "OFF"
+        cmake.definitions["gRPC_BUILD_CSHARP_EXT"] = "ON" if self.options.build_csharp_ext else "OFF"
+        cmake.definitions["gRPC_BUILD_TESTS"] = "OFF"
+
+        # We need the generated cmake/ files (bc they depend on the list of targets, which is dynamic)
+        cmake.definitions["gRPC_INSTALL"] = "ON"
+        # cmake.definitions["CMAKE_INSTALL_PREFIX"] = self._build_subfolder
+
+        # tell grpc to use the find_package versions
+        cmake.definitions["gRPC_ZLIB_PROVIDER"] = "package"
+        cmake.definitions["gRPC_CARES_PROVIDER"] = "package"
+        cmake.definitions["gRPC_RE2_PROVIDER"] = "package"
+        cmake.definitions["gRPC_SSL_PROVIDER"] = "package"
+        cmake.definitions["gRPC_PROTOBUF_PROVIDER"] = "package"
+        cmake.definitions["gRPC_ABSL_PROVIDER"] = "package"
+
+        cmake.definitions["gRPC_BUILD_GRPC_CPP_PLUGIN"] = self.options.build_cpp_plugin
+        cmake.definitions["gRPC_BUILD_GRPC_CSHARP_PLUGIN"] = self.options.build_csharp_plugin
+        cmake.definitions["gRPC_BUILD_GRPC_NODE_PLUGIN"] = self.options.build_node_plugin
+        cmake.definitions["gRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN"] = self.options.build_objective_c_plugin
+        cmake.definitions["gRPC_BUILD_GRPC_PHP_PLUGIN"] = self.options.build_php_plugin
+        cmake.definitions["gRPC_BUILD_GRPC_PYTHON_PLUGIN"] = self.options.build_python_plugin
+        cmake.definitions["gRPC_BUILD_GRPC_RUBY_PLUGIN"] = self.options.build_ruby_plugin
+
+        # see https://github.com/inexorgame/conan-grpc/issues/39
+        if self.settings.os == "Windows":
+            if not self.options["protobuf"].shared:
+                cmake.definitions["Protobuf_USE_STATIC_LIBS"] = "ON"
+            else:
+                cmake.definitions["PROTOBUF_USE_DLLS"] = "ON"
+
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        # tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

Comment:
Hooks will complain about those files not being removed

I did not remove them from the package yet, as there is no components support yet